pkg/compiler/compiler.go

```diff
@@ -357,3 +358,72 @@ func arrayContains(a []string, v string) bool {
 	}
 	return false
 }
+
+func (comp *compiler) flattenFlags() {
+	intFlags := make(map[string]*ast.IntFlags, len(comp.intFlags))
+	intValuesGetter := func(flags *ast.IntFlags) []*ast.Int {
+		return flags.Values
+	}
+	intFlagsGetter := func(val *ast.Int) *ast.IntFlags {
```

Comment:
I'm not sure what you mean. I am accessing `val.Ident`.

`map[F]V` also wouldn't match `intFlags`. In case of `intFlags`, F is `ast.IntFlags` and V is `ast.Int`, whereas `intFlags` is `map[string]*ast.IntFlags`.

Reply:
I found a way to make it work, by replacing my getters by a couple new interface and using a generics map as you suggested. I'll push once I've addressed the other comments.

Reply:
Done with a new implementation of the fourth commit.